flask_resty/spec/spec_declaration.py

```diff
@@ -0,0 +1,9 @@
+class SpecDeclaration(object):
+
+    def update_spec(self, path, spec, **kwargs):
+        """When overridden, gives a chance to update the path fragment of the
+        specification
+        :param dict path: swagger-spec/blob/master/versions/2.0.md#pathItemObject) # noqa
+        :param apispec.APISpec spec: the specification object
+        """
+        pass
```

Comment:
by duck typing you mean doing something like

````py
if callable(getattr(object, 'update_spec'))
````

As I said in another, I find inheritance very handy to express the update_spec mechanism. Not sure how easy would be without inheritance.